### DISCUSSION TITLE
Fabry Perot resonances modeling

### DISCUSSION THREAD

**Discussion by kalosu** [NanoComp/meep#2683] [Q&A]
Hi meep community, 

I want to model a micro meter scale fabry perot cavity and I wonder if meep is the right tool for doing so. 

In essence, I have a cavity with non flat interfaces and I would like to obtain the spectral response for such a cavity. This cavity has a length of roughly 150 microns and a diameter of 125 microns.

Would meep be a suitable option in this case? I want at the end to compute the spectral response, i.e, the peaks positions as a function of wavelength, for different parameters of my cavity design. 

From your experience, would this be possible to model via MEEP or with MPB? 

Any suggestions and comments will be appreciated.

**Comment by smartalecH** [comment]
Traditionally, FDTD _alone_ is often a poor choice for modeling large cavities with high Q's because it can take a long time for the fields to decay. 

_However_, when coupled with a spectral extrapolation technique, it's incredibly efficient. 

For example, you can use meep with Harminv to compute the locations of resonances and their Q's to high precision using just a few oscillations of the cavity (rather than requiring the fields to decay completely). Here's a good tutorial: https://meep.readthedocs.io/en/latest/Python_Tutorials/Resonant_Modes_and_Transmission_in_a_Waveguide_Cavity/#resonant-modes

Similarly, you can compute the entire spectrum from just a few periods of oscillation using a Pade approximant. See #2440 for some examples. (@oskooi we should probably add a tutorial on this at some point).
